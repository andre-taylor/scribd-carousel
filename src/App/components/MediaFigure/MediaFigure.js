import React from 'react';
import PropTypes from 'prop-types';
import { Typography } from '..';
import { Linkable } from '..';

import styles from './MediaFigure.module.css';

export const MediaFigure = ({
  item: { title, author, href, imageSrc, retinaImageSrc },
}) => {
  const RenderFigure = () => {
    return (
      <figure className={styles.mediaContainer}>
        <img
          className={styles.mediaImage}
          alt={title}
          src={imageSrc}
          srcSet={retinaImageSrc}
        />
        <figcaption className={styles.mediaCaption}>
          <Typography
            variant="h3"
            content={title}
            className={styles.mediaTitle}
          />
          {author && (
            <Typography
              variant="h4"
              content={author}
              className={styles.mediaAuthor}
            />
          )}
        </figcaption>
      </figure>
    );
  };

  return (
    <>
      {href ? (
        <Linkable href={href}>{RenderFigure()}</Linkable>
      ) : (
        RenderFigure()
      )}
    </>
  );
};

MediaFigure.propTypes = {
  item: PropTypes.shape({
    author: PropTypes.string,
    href: PropTypes.string,
    title: PropTypes.string.isRequired,
    imageSrc: PropTypes.string.isRequired,
    retinaImageSrc: PropTypes.string,
  }).isRequired,
};

export default MediaFigure;
