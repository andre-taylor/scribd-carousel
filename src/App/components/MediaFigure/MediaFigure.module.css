.mediaContainer {
  padding: 0 6px;
}
.mediaImage {
  background-color: #f8f9fd;
  border: 1px solid #e9edf8;
}
.mediaCaption {
  margin-top: 4px;
}
.mediaTitle {
  font-size: 1.6rem;
  color: #000514;
}
.mediaAuthor {
  font-size: 1.4rem;
  color: #57617a;
}
