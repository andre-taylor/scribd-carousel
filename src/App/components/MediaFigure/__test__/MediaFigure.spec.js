import React from 'react';
import renderer from 'react-test-renderer';

import MediaFigure from '../MediaFigure';

describe('<MediaFigure />', () => {
  it('renders correctly with optional and required props', () => {
    const item = {
      title: 'Media Title',
      author: 'Media Author',
      href: 'https://www.scribd.com',
      imageSrc:
        'https://imgv2-1-f.scribdassets.com/img/word_document/163600930/149x198/25c730f1a0/1556657999?v=1',
      srcSet:
        'https://imgv2-1-f.scribdassets.com/img/word_document/163600930/298x396/6e57b3bff3/1556657999?v=1',
    };

    const component = renderer.create(<MediaFigure item={item} />);
    expect(component).toMatchSnapshot();
  });
  it('renders correctly with required props', () => {
    const item = {
      title: 'Media Title',
      imageSrc:
        'https://imgv2-1-f.scribdassets.com/img/word_document/163600930/149x198/25c730f1a0/1556657999?v=1',
    };

    const component = renderer.create(<MediaFigure item={item} />);
    expect(component).toMatchSnapshot();
  });
});
