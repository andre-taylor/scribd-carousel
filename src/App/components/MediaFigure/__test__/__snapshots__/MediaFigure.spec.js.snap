// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaFigure /> renders correctly with optional and required props 1`] = `
<a
  className="linkableContainer"
  href="https://www.scribd.com"
>
  <figure
    className="mediaContainer"
  >
    <img
      alt="Media Title"
      className="mediaImage"
      src="https://imgv2-1-f.scribdassets.com/img/word_document/163600930/149x198/25c730f1a0/1556657999?v=1"
    />
    <figcaption
      className="mediaCaption"
    >
      <h3
        className="mediaTitle"
      >
        Media Title
      </h3>
      <h4
        className="mediaAuthor"
      >
        Media Author
      </h4>
    </figcaption>
  </figure>
</a>
`;

exports[`<MediaFigure /> renders correctly with required props 1`] = `
<figure
  className="mediaContainer"
>
  <img
    alt="Media Title"
    className="mediaImage"
    src="https://imgv2-1-f.scribdassets.com/img/word_document/163600930/149x198/25c730f1a0/1556657999?v=1"
  />
  <figcaption
    className="mediaCaption"
  >
    <h3
      className="mediaTitle"
    >
      Media Title
    </h3>
  </figcaption>
</figure>
`;
